kmod/core/core.c

```diff
@@ -661,31 +699,26 @@ static int kpatch_write_relocations(struct kpatch_module *kpmod,
        unsigned long core = (unsigned long)kpmod->mod->module_core;
        unsigned long core_size = kpmod->mod->core_size;
 #endif
-	unsigned long src;
+	bool vmlinux = !strcmp(object->name, "vmlinux");
 
 	list_for_each_entry(dynrela, &object->dynrelas, list) {
-		if (!strcmp(object->name, "vmlinux")) {
-			ret = kpatch_verify_symbol_match(dynrela->name,
-							 dynrela->src);
-			if (ret)
-				return ret;
+		if (vmlinux || !dynrela->external) {
+			ret = kpatch_find_object_symbol(vmlinux ? NULL : object->name,
+							dynrela->name,
+							dynrela->sympos,
+							&dynrela->src);
 		} else {
 			/* module, dynrela->src needs to be discovered */
```

Comment:
This comment can be removed, since it doesn't really apply anymore (we are now resolving/discovering dynrela->src in all cases, not just in the case of patching modules)